### PR TITLE
mcfgthread: apply small fixes

### PR DESCRIPTION
**Pull request by ognevny**
@lhmouse that's what I meant :) 
I suppose that tag has ${major}.${minor}-ga.${patch} format, so this pkgver expansion should be always correct. pushed repo checksums while at it

### CODE REVIEW

**Comment by lhmouse** [comment]
Actually I hade been thinking about this for quite a while. For example, in a recent GCC version string, the third field is almost always zero. The branch state is indicated by the second field: 0 for pre-release, and 1+ for minor releases.

In my schedule, tags are named like 'v${abi}.${minor}-${suffix}'. The suffix is a string and it indicates the status of a branch (alpha, beta, rc, ga). There is not necessarily a revision or build number field. But as you have suggested, I can try to keep a revision suffix for ga tags.

**Comment by ognevny** [comment]
so `ga` is for 'stable' releases, right? you can keep your system with beta, rc, etc. but the only version is built here is with ga suffix

**Comment by lhmouse** [comment]
> so `ga` is for 'stable' releases, right? you can keep your system with beta, rc, etc. but the only version is built here is with ga suffix

yes, exactly.

**Comment by Biswa96** [comment]
`_tag='v1.9-ga.1'` seems easier to read and comprehend compared to the pkgver magic.

**Comment by ognevny** [comment]
for me it's better when you need to change pkgver only

**Comment by lhmouse** [comment]
> for me it's better when you need to change pkgver only

I agree with this, like with other packages.

File: mingw-w64-mcfgthread/PKGBUILD
Comment (lhmouse) [inline mingw-w64-mcfgthread/PKGBUILD:1]:
```sh
package_mcfgthread() {
  pkgdesc="An efficient gthread implementation, providing C++11 threading support (devel)"
  depends=("${MINGW_PACKAGE_PREFIX}-${_realname}"-libs)
```

I just saw this today; I think `depends` should be 

```sh
  depends=("${MINGW_PACKAGE_PREFIX}-${_realname}-libs=${pkgver}")
```

**Comment by lhmouse** [comment]
also the license in SPDX form is `LGPL-3.0-or-later WITH GCC-exception-3.1`.

**Comment by ognevny** [comment]
everything is done, changed PR name since it now contains several changes. also bumped pkgrel

**Comment by lhmouse** [comment]
What do `staticlibs` and `strip` mean in `options`?

**Comment by ognevny** [comment]
> What do `staticlibs` and `strip` mean in `options`?

staticlibs: allow staticlibs in package (disabled by default in Arch, but enabled by default in MSYS2)
strip: run strip for each executable/library file (enabled by default)

**Comment by lhmouse** [comment]
> staticlibs: allow staticlibs in package (disabled by default in Arch, but enabled by default in MSYS2) strip: run strip for each executable/library file (enabled by default)

Thanks for the explanation. So they are on by default. The change looks good to me.

**Comment by ognevny** [comment]
> The change looks good to me.

all changes or the exact one?

**Comment by lhmouse** [comment]
I meant this _commit_ looks good to me. 😅

**Comment by ognevny** [comment]
> I meant this _commit_ looks good to me. 😅 

that's what "all changes" mean :)

**Comment by lhmouse** [comment]
thanks; apologies for the confusion.

**Comment by ognevny** [comment]
no problems

**Comment by Biswa96** [comment]
> What do `staticlibs` and `strip` mean in `options`?

`man PKGBUILD` and `man makepkg.conf`

**Comment by lhmouse** [comment]
> > What do `staticlibs` and `strip` mean in `options`?
> 
> 
> 
> `man PKGBUILD` and `man makepkg.conf`

I was actually a bit confused how `staticlib` should work. To me it looks like 'whether staticlibs should be _built_' rather than to _exclude_ them; however for this package there is no way to not build a static lib.

`strip` is for `strip --strip-unneeded` and is actually less confusing. (still a little because it's not `--strip-all`.)

**Pull request merged**